scripts/Diffraction/isis_powder/osiris.py

```diff
@@ -0,0 +1,185 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from isis_powder.abstract_inst import AbstractInst
+from isis_powder.osiris_routines import osiris_advanced_config, osiris_algs, osiris_param_mapping
+from isis_powder.routines import instrument_settings, common_output
+import copy
+import mantid.simpleapi as mantid
+import os
+
+
+class Osiris(AbstractInst):
+
+    def __init__(self, **kwargs):
+        self._drange_sets = {
+            'drange1': osiris_algs.DrangeData('drange1'),
+            'drange2': osiris_algs.DrangeData('drange2'),
+            'drange3': osiris_algs.DrangeData('drange3'),
+            'drange4': osiris_algs.DrangeData('drange4'),
+            'drange5': osiris_algs.DrangeData('drange5'),
+            'drange6': osiris_algs.DrangeData('drange6'),
+            'drange7': osiris_algs.DrangeData('drange7'),
+            'drange8': osiris_algs.DrangeData('drange8'),
+            'drange9': osiris_algs.DrangeData('drange9'),
+            'drange10': osiris_algs.DrangeData('drange10'),
+            'drange11': osiris_algs.DrangeData('drange11'),
+            'drange12': osiris_algs.DrangeData('drange12'),
+        }
+        self._inst_settings = instrument_settings.InstrumentSettings(
+            param_map=osiris_param_mapping.attr_mapping,
+            adv_conf_dict=osiris_advanced_config.get_all_adv_variables(),
```

Comment:
what is adv_conf_dict?

Reply:
They cover a range of parmeters that are required but not necessarily expected to be supplied by the user. I'll add a comment explaining that.